stock_request/models/stock_request_abstract.py

```diff
@@ -37,12 +37,17 @@ def _compute_product_qty(self):
 
     name = fields.Char("Name", copy=False, required=True, readonly=True, default="/")
     warehouse_id = fields.Many2one(
-        "stock.warehouse", "Warehouse", ondelete="cascade", required=True
+        comodel_name="stock.warehouse",
+        string="Warehouse",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```

Comment:
I think this should go with new multi-company features (like check_company=True).